lib/fine-types/test/data/Cardano/Ledger/Allegra/Tx.fine

```diff
@@ -0,0 +1,60 @@
+module Tx where
+
+import Address(Addr, Addr_rwd, Ix, Slot);
+import Crypto(VKey, Sig, ScriptHash, Script, KeyHash_G);
+import PParams(Coin, Epoch, PParamsUpdate);
+import Delegation(DCert);
+import Script(Script^native_v2);
+
+{-----------------------------------------------------------------------------
+    Transactions
+    Mary spec, Figure 5
+------------------------------------------------------------------------------}
+-- Abstract types
+
+TxId                = _ ; --^ transaction identifier
+Metadatum           = _ ; --^ metadatum
+AuxiliaryDataHash   = _ ; --^ auxiliary data hash
+
+-- Derived types
+
+TxIn                = TxId × Ix; --^ transaction input
+TxOut               = Addr × Coin; --^ transaction output
+UTxO                = TxIn ↦ TxOut; --^ unspent tx outputs
+Wdrl                = Addr_rwd ↦ Coin; --^ reward withdrawal
+Metadata            = ℕ ↦ Metadatum; --^ metadata
+
```

Comment:
```suggestion
Script              = Script^native_ph2;
```

That's how the Mary figure tells it. 🤷

Reply:
Script              = Script^native_v2; you mean ?

Reply:
Yes. 😱 Sorry.